FFmpegInterop/SampleProvider.cpp

```diff
@@ -0,0 +1,415 @@
+
+		switch (codecPar->codec_type)
+		{
+		case AVMEDIA_TYPE_AUDIO:
+		{
+			IAudioEncodingProperties audioEncProp{ encProp.as<IAudioEncodingProperties>() };
+			audioEncProp.SampleRate(codecPar->sample_rate);
+			audioEncProp.ChannelCount(codecPar->channels);
+			audioEncProp.Bitrate(static_cast<uint32_t>(codecPar->bit_rate));
+			audioEncProp.BitsPerSample(static_cast<uint32_t>(max(codecPar->bits_per_coded_sample, codecPar->bits_per_raw_sample)));
+
+			if (setFormatUserData && codecPar->extradata != nullptr && codecPar->extradata_size > 0)
+			{
+				// Set the format user data
+				IAudioEncodingPropertiesWithFormatUserData audioEncPropWithFormatUserData{ encProp.as<IAudioEncodingPropertiesWithFormatUserData>() };
+				audioEncPropWithFormatUserData.SetFormatUserData({ codecPar->extradata, codecPar->extradata + codecPar->extradata_size });
+			}
+
+			MediaPropertySet properties{ audioEncProp.Properties() };
+
+			if (GUID subtype{ unbox_value<GUID>(properties.Lookup(MF_MT_SUBTYPE)) }; subtype == MFAudioFormat_PCM || subtype == MFAudioFormat_Float)
+			{
+				properties.Insert(MF_MT_ALL_SAMPLES_INDEPENDENT, PropertyValue::CreateUInt32(true));
+				properties.Insert(MF_MT_COMPRESSED, PropertyValue::CreateUInt32(false));
+			}
+			else
+			{
+				properties.Insert(MF_MT_ALL_SAMPLES_INDEPENDENT, PropertyValue::CreateUInt32(false));
+				properties.Insert(MF_MT_COMPRESSED, PropertyValue::CreateUInt32(true));
+			}
+			
+			break;
+		}
+
+		case AVMEDIA_TYPE_VIDEO:
+		{
+			IVideoEncodingProperties videoEncProp{ encProp.as<IVideoEncodingProperties>() };
+			videoEncProp.Height(codecPar->height);
+			videoEncProp.Width(codecPar->width);
+			videoEncProp.Bitrate(static_cast<uint32_t>(codecPar->bit_rate));
+
+			if (codecPar->sample_aspect_ratio.num > 0 && codecPar->sample_aspect_ratio.den != 0)
+			{
+				MediaRatio pixelAspectRatio{ videoEncProp.PixelAspectRatio() };
+				pixelAspectRatio.Numerator(codecPar->sample_aspect_ratio.num);
+				pixelAspectRatio.Denominator(codecPar->sample_aspect_ratio.den);
+			}
+
+			if (m_stream->avg_frame_rate.num != 0 || m_stream->avg_frame_rate.den != 0)
+			{
+				MediaRatio frameRate{ videoEncProp.FrameRate() };
+				frameRate.Numerator(m_stream->avg_frame_rate.num);
+				frameRate.Denominator(m_stream->avg_frame_rate.den);
+			}
+
+			if (setFormatUserData && codecPar->extradata != nullptr && codecPar->extradata_size > 0)
+			{
+				// Set the format user data
+				IVideoEncodingProperties2 videoEncProp2{ encProp.as<IVideoEncodingProperties2>() };
+				videoEncProp2.SetFormatUserData({ codecPar->extradata, codecPar->extradata + codecPar->extradata_size });
+			}
+
+			MediaPropertySet properties{ videoEncProp.Properties() };
+
+			properties.Insert(MF_MT_ALL_SAMPLES_INDEPENDENT, PropertyValue::CreateUInt32(false));
+			properties.Insert(MF_MT_COMPRESSED, PropertyValue::CreateUInt32(true));
+			properties.Insert(MF_MT_PIXEL_ASPECT_RATIO, PropertyValue::CreateUInt64(Pack2UINT32AsUINT64(1, 1)));

```

Comment:
>properties.Insert(MF_MT_PIXEL_ASPECT_RATIO, PropertyValue::CreateUInt64(Pack2UINT32AsUINT64(1, 1))); [](start = 3, length = 100)

It looks like this would override the pixelAspectRatio set above - can this be removed?